### PR TITLE
feat(review-workflow): Stages

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

- Creates the Stage content type for Review Workflows.

### Why is it needed?

- So workflows can have and populate their own stages.

### How to test it?

- See api tests.

### CODE REVIEW

**Review by nathan-pichon — APPROVED** [review]
I have applied everything that was asked in the previous Workflow content type PR. Everything looks good. Needs another approve as I'm one of the developers of this PR.

File: packages/core/admin/ee/server/controllers/workflows/stages/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+const { getService } = require('../../../utils');
+
+module.exports = {
+  /**
+   * List all stages
+   * @param {import('koa').BaseContext} ctx - koa context
+   */
+  async find(ctx) {
+    const { id } = ctx.params;
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/controllers/workflows/stages/index.js:11]: I think this should be workflowId instead
```suggestion
    const { workflow_id } = ctx.params;
```

File: packages/core/admin/ee/server/content-types/workflow-stage/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+module.exports = {
+  schema: {
+    collectionName: 'strapi_workflows_stages',
```
Comment (gu-stav) [inline packages/core/admin/ee/server/content-types/workflow-stage/index.js:5]: Should this be 
```suggestion
    collectionName: 'strapi_workflow_stages',
```

?

File: packages/core/admin/ee/server/controllers/workflows/index.js [hunk 3]
```diff
@@ -21,9 +24,10 @@ module.exports = {
    */
   async findOne(ctx) {
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/controllers/workflows/index.js:25]: Maybe we could keep the consistency between service and controller and rename this findById too, wdyt?

File: packages/core/admin/ee/server/services/__tests__/stages.test.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  describe('find', () => {
+    test('Should call entityService with the right model UID and ID', async () => {
+      stagesService.find({ workflowId: 1 });
+
+      expect(entityServiceMock.findOne).not.toBeCalled();
+      expect(entityServiceMock.findMany).toBeCalled();
+      expect(entityServiceMock.findMany).toBeCalledWith(STAGE_MODEL_UID, {
+        filter: { workflow: 1 },
+      });
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/services/__tests__/stages.test.js:53]: I do not understand how is this working if the workflow Id is not passed correctly. 🤔

File: packages/core/admin/ee/server/controllers/workflows/stages/index.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+   * List all stages
+   * @param {import('koa').BaseContext} ctx - koa context
+   */
+  async find(ctx) {
+    const { workflow_id: workflowId } = ctx.params;
+    const { populate } = ctx.query;
+    const stagesService = getService('stages');
+
+    const data = await stagesService.find({
+      workflowId: id,
```
Comment (nathan-pichon) [inline packages/core/admin/ee/server/controllers/workflows/stages/index.js:16]:
```suggestion
      workflowId
```

File: packages/core/admin/ee/server/content-types/workflow-stage/index.js (hunk 2 above)
Comment (nathan-pichon) [reply packages/core/admin/ee/server/content-types/workflow-stage/index.js:5]: This collection will store all the stages for all the workflows, shouldn't we keep it plural ?

File: packages/core/admin/ee/server/services/__tests__/stages.test.js (hunk 4 above)
Comment (nathan-pichon) [reply packages/core/admin/ee/server/services/__tests__/stages.test.js:53]: I have added one more stage in the integration test to be used for findById controller.

**Comment by Marc-Roig** [comment]
LGTM, I can not approve as I created the PR originally.

**Review by jhoward1994 — APPROVED** [review]
LGTM 👍🏻 I had one really minor comment

**Pull request merged**